File: frontend/src/service/conversation.service.ts
import {EntityService} from "./entity.service";
import {HttpClient} from "@angular/common/http";
import {Injectable} from "@angular/core";
import {Conversation} from "../model/conversation";
import {Observable} from "rxjs";

@Injectable({
  providedIn: 'root'
})
export class ConversationService extends EntityService<Conversation> {
  constructor(http: HttpClient) {
    super(http, "conversation");
  }

  getConversationsByStudentId(studentId: number): Observable<Conversation[]> {
    return this.http.get<Conversation[]>(`${this.apiBackendUrl}/${this.entityName}/get-conversations-by-student-id/${studentId}`);
  }

  getConversationsByProfessorId(professorId: number): Observable<Conversation[]> {
    return this.http.get<Conversation[]>(`${this.apiBackendUrl}/${this.entityName}/get-conversations-by-professor-id/${professorId}`);
  }
}

File: backend/src/main/java/arbremojo/backend/service/ConversationService.java
package arbremojo.backend.service;

import arbremojo.backend.model.Conversation;
import arbremojo.backend.repository.ConversationRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class ConversationService extends EntityService<Conversation, ConversationRepository> {

    @Autowired
    public ConversationService(ConversationRepository entityRepository) {
        super(entityRepository);
    }

    @Override
    public Integer deleteEntityById(Integer id) {
        return entityRepository.deleteEntityById(id);
    }

    public List<Conversation> getConversationsByProfessorId(Integer professorId) {
        return entityRepository.getConversationsByProfessorId(professorId);
    }

    public List<Conversation> getConversationsByStudentId(Integer studentId) {
        return entityRepository.getConversationsByStudentId(studentId);
    }
}
